lacework/provider.go

```diff
@@ -162,8 +169,16 @@ func providerConfigure(_ context.Context, d *schema.ResourceData) (interface{},
 		}
 	}
 
-	if account != "" && key != "" && secret != "" {
-		apiOpts = append(apiOpts, api.WithApiKeys(key, secret))
+	// authentication via environment variables or static credentials
+	if validStaticCredentials(account, key, secret, token) {
+		if token != "" {
+			apiOpts = append(apiOpts, api.WithToken(token))
```

Comment:
Noob question. If token is `nil` does it still call `api.WithToken(token)`? I assume that logic is handled somewhere else and this codepath doesn't have to handle that logic.

Reply:
You assumption is correct, Terraform has out-of-the-box schema validation, so every argument (token, key, account, etc.) has already been validated safely. The line you are looking at for the `token` string is:
```go
139	token = d.Get("api_token").(string)
```